src/app/rom.cc

```diff
@@ -148,24 +124,29 @@ CompressionPiece* SplitCompressionPiece(CompressionPiece* piece, int mode) {
   return new_piece;
 }
 
-uint CreateCompressionString(CompressionPiece* start, uchar* output, int mode) {
+Bytes CreateCompressionString(std::shared_ptr<CompressionPiece>& start,
+                                int mode) {
   uint pos = 0;
-  CompressionPiece* piece = start;
+  auto piece = start;
+  Bytes output;
 
   while (piece != nullptr) {
     // Normal header
     if (piece->length <= kMaxLengthNormalHeader) {
-      output[pos++] = BUILD_HEADER(piece->command, piece->length);
+      output.push_back(BUILD_HEADER(piece->command, piece->length));
+      pos++;
     } else {
       if (piece->length <= kMaxLengthCompression) {
-        output[pos++] = (7 << 5) | ((uchar)piece->command << 2) |
-                        (((piece->length - 1) & 0xFF00) >> 8);
+        output.push_back((7 << 5) | ((uchar)piece->command << 2) |
```

Comment:
Use `#define` to label magic constant `7 << 5`

Reply:
Donezo.